doc/cookbook/time_windows.md

```diff
@@ -0,0 +1,189 @@
+# Processing Time-Windowed Data
+
+If you are analyzing data that is changing over time, chances are that you will want to perform some sort of analysis on "the last two weeks of data," "January's data," or some other moving or static time window of data.  There are a few different ways of doing these types of analyses in Pachyderm, depending on your use case.  We recommend one of the following patterns:
+
+1. [Fixed time window directory structures](#fixed-time-window-directory-structures) - for rigid, fixed time windows, such as months (Jan, Feb, etc.) or days (01-01-17, 01-02-17, etc.).
+
+2. [Rolling time window directory structures](#rolling-time-window-directory-structures) - for rolling time windows of data, such as three day windows or two week windows. 
+
+3. [Shuffled moving time windows](#shuffled-moving-time-windows) - for updating a constanly moving time-windowed data set.
+
+## Fixed time window directory structures
+
+As further discussed in [Creating Analysis Pipelines](http://docs.pachyderm.io/en/latest/fundamentals/creating_analysis_pipelines.html) and [Distributed Computing](http://docs.pachyderm.io/en/latest/fundamentals/distributed_computing.html), the basic unit of data partitioning in Pachyderm is a "datum" which is defined by a glob pattern. When analyzing data within fixed time windows (e.g., corresponding to fixed calendar times/dates), we recommend organizing your data repositories such that each of the time windows that you are going to analyze corresponds to a separate file or directory in your repository. By doing this, you will be able to:
+
+- Analyze each time window in parallel.
+- Only re-process data within a time window when that data, or a corresponding data pipeline, changes.
+
+For example, if you have monthly time windows of CSV sales data that need to be analyzed, you could create a `sales` data repository and structure it like:
+
+```
+sales
+├── January.csv
+├── February.csv
+├── March.csv
+└── April.csv
+```
+
+When you run a pipeline with a glob pattern of `/*` on `sales`, each of the months of sales data is processed in parallel (if possible).  Further, when you add a new month (e.g., `May.csv`), only that new datum, `May.csv`, will be processed.
+
+In some cases, you may also have more than one file associated with a particular time window or more than one timeframe that is relevant (e.g., both months and days).  In these cases, you would want to create a directory structure like:
+
+```
+sales
+├── January
+|   ├── online_sales.csv
+|   └── in_store_sales.csv
+├── February
+|   ├── online_sales.csv
+|   └── in_store_sales.csv
+└── March
+    ├── online_sales.csv
+    └── in_store_sales.csv
+```
+
+or, in the case when you need two time scales (months AND days):
+
+```
+sales
```

Comment:
This example is actually the core use case and it's has nothing to do "two time scales". Time windows are all about collecting data at interval A, but wanted to get aggregates over window B. The most common time window being collecting one file per day, but wanting to keep month totals building up. 

Take out the first two examples in code blocks here. having one file per month is not a good use case for pachyderm because that would be an append-based workload and we dont want that. The 2nd example is also just kind of weak because it implies that you're just adding one new file (e.g. online_sales and in_store_sales) for each month and that's not the point of time windowing. 

Happy to discuss live if I'm not describing this clearly.